### ISSUE TITLE
exec-watch stops working after a while

### ISSUE THREAD

**Issue by SHSE** [coreos/etcd#2183]
Consider the following systemd unit:
```
[Unit]
Requires=docker.service etcd.service
After=docker.service etcd.service

[Service]
ExecStartPre=/usr/bin/docker run --rm -v /opt/bin:/opt/bin ibuildthecloud/systemd-docker

ExecStart=/bin/sh -c "\
   export DB_HOST=`etcdctl get /services/db/host` && \
   if [ ! $DB_HOST ]; then exit 1; fi; \
   etcdctl exec-watch --recursive /services/db/host -- systemctl restart %n & \
   /opt/bin/systemd-docker run --rm --name %p busybox"

Restart=always
RestartSec=10
Type=notify
NotifyAccess=all
TimeoutStartSec=120
TimeoutStopSec=15

[Install]
WantedBy=multi-user.target
```

It launches `etcdctl exec-watch` in the background and the docker container in the foreground. 

The idea is to restart the service when something was changed in the etcd key.

After a while (about a week, maybe a couple of days) it has stopped reacting on a change in the key. The `etcdctl exec-watch` process is alive.

**Comment by xiang90** [comment]
Can you please provide the log, etcdctl and etcd versions?

**Comment by xiang90** [comment]
@SHSE I think this should be fixed in etcd 2.0. I am closing this now. Reopen it if you still have a problem.
